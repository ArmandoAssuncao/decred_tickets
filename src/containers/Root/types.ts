import { NavigationDispatch } from 'react-navigation'

export interface IActionCreators {
}

export interface IConnectedProps {
}

export interface IProps extends IActionCreators, IConnectedProps {
}

export interface IState {
}

export interface IScreenProps {
    screenProps?: {
        resetNavigateTo?: (routeName: string, dispatch: NavigationDispatch) => void
    }
}
