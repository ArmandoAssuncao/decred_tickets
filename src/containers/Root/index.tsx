import Navigator from 'pages'
import React, { Component } from 'react'
import { StatusBar, View, } from 'react-native'
import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux'
import { bindActionCreators } from 'redux'
import { IAppState } from 'store/types'

import styles from './styles'
import { IActionCreators, IConnectedProps, IProps, IState } from './types'
import { resetNavigateTo } from 'helpers/navigate'

const mapStateToProps: MapStateToProps<IConnectedProps, IProps, IAppState> = (_state) => ({
})
const mapDispatchToProps: MapDispatchToProps<IActionCreators, IProps> = (dispatch) =>
    bindActionCreators(
        {},
        dispatch,
    )

class Root extends Component<IProps, IState> {

    render(): JSX.Element {
        return (
            <View style={styles.container}>
                <StatusBar hidden={false} barStyle="light-content" />
                <Navigator
                    screenProps={{
                        resetNavigateTo,
                    }}
                />
            </View>
        )
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Root)
