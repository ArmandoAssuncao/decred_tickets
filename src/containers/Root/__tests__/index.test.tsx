import 'rxjs'
import React from 'react'
import 'react-native'
import { shallowStore } from 'test/enzyme'
import Root from '../index'

describe('Root component', () => {
    const store = {}
    const mockProps = {
        screenProps: {
            resetNavigateTo: jest.fn(),
        },
    }

    it('should match snapshot', () => {
        const wrapper = shallowStore(<Root {...mockProps} />, store)
        expect(wrapper).toMatchSnapshot()
    })
})
