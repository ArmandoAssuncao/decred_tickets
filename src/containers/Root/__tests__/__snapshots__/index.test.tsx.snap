// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root component should match snapshot 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <StatusBar
    animated={false}
    barStyle="light-content"
    hidden={false}
    showHideTransition="fade"
  />
  <NavigationContainer
    screenProps={
      Object {
        "resetNavigateTo": [Function],
      }
    }
  />
</Component>
`;
